src/Pact/Analyze/Analyze.hs

```diff
@@ -484,21 +478,31 @@ instance HasAnalyzeEnv QueryEnv   where analyzeEnv = qeAnalyzeEnv
 
 class (MonadError AnalyzeFailure m) => Analyzer m term where
   analyze  :: (Show a, SymWord a) => term a -> m (S a)
+
+class Analyzer m term => ObjectAnalyzer m term where
   analyzeO :: term Object -> m Object
```

Comment:
Do we need the subclass here? Since every implementor of `Analyzer` also implements `ObjectAnalyzer`, and the two functions are very related, it seems like we can have two methods on `Analyzer` instead of adding another typeclass?

Reply:
`Invariant` does not implement `analyzeO`

Reply:
Missed that. Sounds good!